PR are not checked

- Gitea version (or commit ref): 1.13.1 
- Git version: 2.20.1
- Operating system: Debian 9.12
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ x ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ x ] No
- Log gist:

```
panic: interface conversion: interface is nil, not queue.UniqueQueue
goroutine 21890642 [running]:
code.gitea.io/gitea/modules/queue.(*PersistableChannelUniqueQueue).Has(0xc0041fac40, 0x232c9c0, 0xc00945e760, 0x2, 0xc00f51ca90, 0xc00835dc30)
        /var/git/go/src/code.gitea.io/gitea/modules/queue/unique_queue_disk_channel.go:152 +0x9d
code.gitea.io/gitea/services/pull.checkAndUpdateStatus(0xc000d52300)
        /var/git/go/src/code.gitea.io/gitea/services/pull/check.go:59 +0x97
code.gitea.io/gitea/services/pull.handle(0xc0103c5e20, 0x1, 0x1)
        /var/git/go/src/code.gitea.io/gitea/services/pull/check.go:227 +0x486
code.gitea.io/gitea/modules/queue.NewChannelUniqueQueue.func1(0xc001063040, 0x14, 0x14)
        /var/git/go/src/code.gitea.io/gitea/modules/queue/unique_queue_channel.go:59 +0x13d
code.gitea.io/gitea/modules/queue.(*WorkerPool).doWork(0xc003c6b180, 0x467d840, 0xc00f0fc780)
        /var/git/go/src/code.gitea.io/gitea/modules/queue/workerpool.go:344 +0x344
code.gitea.io/gitea/modules/queue.(*WorkerPool).addWorkers.func1(0xc003c6b180, 0x467d840, 0xc00f0fc780)
        /var/git/go/src/code.gitea.io/gitea/modules/queue/workerpool.go:238 +0x43
created by code.gitea.io/gitea/modules/queue.(*WorkerPool).addWorkers
        /var/git/go/src/code.gitea.io/gitea/modules/queue/workerpool.go:237 +0x56
```

```
panic: interface conversion: interface is nil, not queue.UniqueQueue

goroutine 20300 [running]:
code.gitea.io/gitea/modules/queue.(*PersistableChannelUniqueQueue).Has(0xc004e89110, 0x232c9c0, 0xc00bc42a90, 0x2, 0xc0050fc340, 0xc0051b3c30)
        /var/git/go/src/code.gitea.io/gitea/modules/queue/unique_queue_disk_channel.go:152 +0x9d
code.gitea.io/gitea/services/pull.checkAndUpdateStatus(0xc004fc0d00)
        /var/git/go/src/code.gitea.io/gitea/services/pull/check.go:59 +0x97
code.gitea.io/gitea/services/pull.handle(0xc015f1e890, 0x1, 0x1)
        /var/git/go/src/code.gitea.io/gitea/services/pull/check.go:227 +0x486
code.gitea.io/gitea/modules/queue.NewChannelUniqueQueue.func1(0xc00e0e4280, 0x14, 0x14)
        /var/git/go/src/code.gitea.io/gitea/modules/queue/unique_queue_channel.go:59 +0x13d
code.gitea.io/gitea/modules/queue.(*WorkerPool).doWork(0xc0005bed00, 0x467d7c0, 0xc007d3fe80)
        /var/git/go/src/code.gitea.io/gitea/modules/queue/workerpool.go:344 +0x344
code.gitea.io/gitea/modules/queue.(*WorkerPool).addWorkers.func1(0xc0005bed00, 0x467d7c0, 0xc007d3fe80)
        /var/git/go/src/code.gitea.io/gitea/modules/queue/workerpool.go:238 +0x43
created by code.gitea.io/gitea/modules/queue.(*WorkerPool).addWorkers
        /var/git/go/src/code.gitea.io/gitea/modules/queue/workerpool.go:237 +0x56
```

## Description

All PR are blocked in stale state. 
When we force pr_patch_checker with on process. We get previous error. A restart is needed
First log when we set 1 process to 5minutes, second when unlimited time.

--- a/modules/queue/unique_queue_disk_channel.go
+++ b/modules/queue/unique_queue_disk_channel.go
@@ -149,6 +149,11 @@ func (q *PersistableChannelUniqueQueue) Has(data Data) (bool, error) {
 	if err != nil || has {
 		return has, err
 	}
+	q.lock.Lock()
+	defer q.lock.Unlock()
+	if q.internal == nil {
+		return false, nil
+	}
 	return q.internal.(UniqueQueue).Has(data)
 }
 

